components/api-resource-mgt/org.wso2.carbon.identity.api.resource.mgt/src/main/java/org/wso2/carbon/identity/api/resource/mgt/dao/impl/CacheBackedApiResourceMgtDAO.java

```diff
@@ -0,0 +1,284 @@
+
+        if (apiResource != null) {
+            log.debug("Entry fetched from DB for API Resource " + identifier + ". Updating cache");
+            apiResourceCacheByIdentifier.addToCache(cacheKey, new ApiResourceCacheEntry(apiResource), tenantId);
+            if (apiResource.getId() != null) {
+                ApiResourceIdCacheKey apiResourceIdCacheKey = new ApiResourceIdCacheKey(apiResource.getId());
+                apiResourceCacheById.addToCache(apiResourceIdCacheKey,
+                        new ApiResourceCacheEntry(apiResource), tenantId);
+            }
+        } else {
+            log.debug("Entry for API Resource " + identifier + " not found in cache or DB");
+        }
+
+        return apiResource;
+    }
+
+    @Override
+    public void updateApiResource(ApiResource apiResource, List<Scope> addedScopes, List<String> removedScopes,
+                                  Integer tenantId) throws ApiResourceManagementException {
+
+        clearApiResourceCache(apiResource.getIdentifier(), apiResource.getId(), tenantId);
+        apiResourceManagementDAO.updateApiResource(apiResource, addedScopes, removedScopes, tenantId);
+    }
+
+    @Override
+    public void deleteApiResourceById(String apiId, Integer tenantId) throws ApiResourceManagementException {
+
+        clearApiResourceCache(null, apiId, tenantId);
+        apiResourceManagementDAO.deleteApiResourceById(apiId, tenantId);
+    }
+
+    @Override
+    public boolean isScopeExistByName(String name, Integer tenantId) throws ApiResourceManagementException {
+
+        return apiResourceManagementDAO.isScopeExistByName(name, tenantId);
+    }
+
+    @Override
+    public boolean isScopeExistById(String scopeId) throws ApiResourceManagementException {
+
+        return apiResourceManagementDAO.isScopeExistById(scopeId);
+    }
+
+    @Override
+    public Scope getScopeByNameAndTenantId(String name, Integer tenantId) throws ApiResourceManagementException {
+
+        return apiResourceManagementDAO.getScopeByNameAndTenantId(name, tenantId);
+    }
+
+    @Override
+    public Scope getScopeByNameTenantIdApiId(String name, Integer tenantId, String apiId)
+            throws ApiResourceManagementException {
+
+        return apiResourceManagementDAO.getScopeByNameTenantIdApiId(name, tenantId, apiId);
+    }
+
+    @Override
+    public List<Scope> getScopesByTenantId(Integer tenantId, List<ExpressionNode> expressionNodes)
+            throws ApiResourceManagementException {
+
+        return apiResourceManagementDAO.getScopesByTenantId(tenantId, expressionNodes);
+    }
+
+    @Override
+    public void addScopes(List<Scope> scopes, String apiId, Integer tenantId) throws ApiResourceManagementException {
+
+        clearApiResourceCache(null, apiId, tenantId);
+        apiResourceManagementDAO.addScopes(scopes, apiId, tenantId);
+    }
+
+//    @Override
+//    public void patchScopes(String apiId, List<Scope> addedScopes, List<String> removedScopes, Integer tenantId)
```

Comment:
Let's remove this if not needed

Reply:
https://github.com/wso2/carbon-identity-framework/pull/4854/commits/8dd16447d6000412e3ff2dd1b2a9ed78249f2d13